knative-operator/pkg/webhook/knativeeventing/webhook_validating.go

```diff
@@ -0,0 +1,130 @@
+package knativeeventing
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/openshift-knative/serverless-operator/knative-operator/pkg/common"
+	webhookutil "github.com/openshift-knative/serverless-operator/knative-operator/pkg/webhook/util"
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	eventingv1alpha1 "knative.dev/eventing-operator/pkg/apis/eventing/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission/builder"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission/types"
+)
+
+// Creates a new validating KnativeEventing Webhook
+func ValidatingWebhook(mgr manager.Manager) (webhook.Webhook, error) {
+	common.Log.Info("Setting up validating webhook for KnativeEventing")
+	return builder.NewWebhookBuilder().
+		Name("validating.knativeeventing.openshift.io").
+		Validating().
+		Operations(admissionregistrationv1beta1.Create, admissionregistrationv1beta1.Update).
+		WithManager(mgr).
+		ForType(&eventingv1alpha1.KnativeEventing{}).
+		Handlers(&KnativeEventingValidator{}).
+		Build()
+}
+
+// KnativeEventingValidator validates KnativeEventing CR's
+type KnativeEventingValidator struct {
+	client  client.Client
+	decoder types.Decoder
+}
+
+// Implement admission.Handler so the controller can handle admission request.
+var _ admission.Handler = (*KnativeEventingValidator)(nil)
+
+// What makes us a webhook
+func (v *KnativeEventingValidator) Handle(ctx context.Context, req types.Request) types.Response {
+	ke := &eventingv1alpha1.KnativeEventing{}
+
+	err := v.decoder.Decode(req, ke)
+	if err != nil {
+		return admission.ErrorResponse(http.StatusBadRequest, err)
+	}
+
+	allowed, reason, err := v.validate(ctx, ke)
+	if err != nil {
+		return admission.ErrorResponse(http.StatusInternalServerError, err)
+	}
+	return admission.ValidationResponse(allowed, reason)
+}
+
+// KnativeEventingValidator checks for a minimum OpenShift version
+func (v *KnativeEventingValidator) validate(ctx context.Context, ke *eventingv1alpha1.KnativeEventing) (allowed bool, reason string, err error) {
+	log := common.Log.WithName("validate")
+	stages := []func(context.Context, *eventingv1alpha1.KnativeEventing) (bool, string, error){
+		v.validateNamespace,
+		v.validateVersion,
+		v.validateLoneliness,
```

Comment:
I'd really like to figure out how to refactor these into a single pkg that works for both KE and KS. We can do that later, of course.